internal/server/api/server_handlers.go

```diff
@@ -46,106 +44,112 @@ func (s *Server) IndexHandler(w http.ResponseWriter, r *http.Request) {
 
 	_, err = w.Write([]byte(html))
 	if err != nil {
-		log.Printf("an error occured writing to browser: %v", err)
+		s.logger.Info(fmt.Sprintf("an error occured writing to browser: %v", err))
 	}
 }
 
 func getAllMetrics(s *Server) string {
 	html := "<h3>Gauge:</h3>"
-	for mName, mValue := range s.store.Gauge {
+	for mName, mValue := range s.store.MemStore.Gauge {
 		html += (mName + ":" + strconv.FormatFloat(mValue, 'f', -1, 64) + "<br>")
 	}
 	html += "<h3>Counter:</h3>"
-	for mName, mValue := range s.store.Counter {
+	for mName, mValue := range s.store.MemStore.Counter {
 		html += (mName + ":" + strconv.FormatInt(mValue, 10) + "<br>")
 	}
 	return html
 }
 
 func (s *Server) GetHandler(lg *zap.Logger) http.HandlerFunc {
 	return func(w http.ResponseWriter, r *http.Request) {
-		mType := chi.URLParam(r, "mtype")
-		mName := chi.URLParam(r, "mname")
-		mValue, err := s.getSingleMetric(mType, mName)
-		if err != nil {
-			http.Error(w, err.Error(), http.StatusNotFound)
-			return
-		}
-		t := s.tpl
-		var doc bytes.Buffer
-		err = t.Execute(&doc, mValue)
-		if err != nil {
-			log.Printf("an error occured processing template data: %v", err)
-			w.WriteHeader(http.StatusInternalServerError)
-			return
-		}
-		w.Header().Set(ContentType, "text/html")
-		w.Header().Set("charset", "utf-8")
-		w.WriteHeader(http.StatusOK)
-
-		html := doc.String()
+		lg.Info("Request URI: " + r.RequestURI)
+		lg.Info("Request method: " + r.Method)
 
-		_, err = w.Write([]byte(html))
-		if err != nil {
-			log.Printf("an error occured writing to browser: %v", err)
+		requestedJSON := false
+		contentType := r.Header.Get("Content-Type")
+		if contentType == applicationJSON {
+			requestedJSON = true
 		}
-	}
-}
 
-func (s *Server) GetHandlerJSON(lg *zap.Logger) http.HandlerFunc {
-	return func(w http.ResponseWriter, r *http.Request) {
-		body, err := io.ReadAll(r.Body)
-		if err != nil {
-			panic(err)
-		}
-
-		w.Header().Set(ContentType, applicationJSON)
+		if requestedJSON {
+			body, err := io.ReadAll(r.Body)
+			if err != nil {
+				lg.Info("error reading request body", zap.Error(err))
+			}
 
-		var m storage.Metrics
+			w.Header().Set(ContentType, applicationJSON)
 
-		err = json.Unmarshal(body, &m)
-		if err != nil {
-			lg.Info("Got error decoding JSON request")
-			w.WriteHeader(http.StatusBadRequest)
-			return
-		}
+			var m storage.Metrics
 
-		mValue, err := s.getSingleMetric(m.MType, m.ID)
-		if err != nil {
-			lg.Info(m.MType + " value doesn't exist for ID: " + m.ID)
-			w.WriteHeader(http.StatusNotFound)
-			return
-		} else {
-			if mValue == "" {
-				w.WriteHeader(http.StatusNotFound)
+			err = json.Unmarshal(body, &m)
+			if err != nil {
+				w.WriteHeader(http.StatusBadRequest)
 				return
 			}
 
-			w.WriteHeader(http.StatusOK)
-
-			var metric = storage.Metrics{}
+			mValue, err := s.getSingleMetric(m.MType, m.ID)
+			if err != nil {
+				lg.Info(m.MType + " value doesn't exist for ID: " + m.ID)
```

Comment:
1.

```suggestion
				lg.Infof(m.MType + " value doesn't exist for ID: " + m.ID)
```

2. Логируются только ошибки привоящие к статус-коду >= 500

Reply:
Убрал